src/julia.h

```diff
@@ -1492,6 +1495,8 @@ STATIC_INLINE void jl_eh_restore_state(jl_handler_t *eh)
     }
 #endif
     JL_SIGATOMIC_END();
+    jl_finalizers_enable(eh->finalizers_enabled);
+    jl_gc_enable(eh->gc_enabled);
```

Comment:
this changes the behavior of the test in `misc` for gc being disabled after `try; gc_enable(false); end`. I'm actually inclined to think the the new behavior is better (and fix the test / document that `gc_enable` is scoped to the nearest try/catch blocks), but I'm open to counter-arguments.